$this->assertValidationErrors error: Cannot call constructor

<!--- Provide a general summary of the issue in the Title above -->

### Preconditions
<!--- Provide a more detailed information of environment you use -->
<!--- Bundle version, tag, HEAD, etc., PHP version, etc.. -->
I am using Symfony 3.4 with the following bundles:
```
liip/functional-test-bundle 3.1.0 3.1.0 This bundles provides additional functional test-cases fo...
liip/test-fixtures-bundle   1.0.1 1.0.1 This bundles enables efficient loading of Doctrine fixtur...
```

### Steps to reproduce
<!--- Provide a set of unambiguous steps to reproduce this bug include code, if relevant  -->
1. Create a class that extends `Liip\FunctionalTestBundle\Test\WebTestCase`
2. Use the following method: `$this->assertValidationErrors([], $client->getContainer())`

### Expected result
<!--- Tell us what should happen -->
1. The list of validation errors should be shown

### Actual result
<!--- Tell us what happens instead -->
My code:
```
    public function testRegisterNoInput()
    {
        $client  = $this->makeClient();
        $crawler = $client->request(
            'POST',
            '/api/patients/register'
        );

        $this->assertIsBadRequestRestApiError($client, ApiResponse::CODE_INVALID_INPUT);
        $this->assertValidationErrors([], $client->getContainer());
    }
```

```
PHPUnit 8.2.4 by Sebastian Bergmann and contributors.

Testing Project Test Suite
E                                                                   1 / 1 (100%)

Time: 290 ms, Memory: 36.25 MB

There was 1 error:

1) Tests\AppBundle\Controller\Api\PatientControllerTest::testRegisterNoInput
Error: Cannot call constructor

/vendor/liip/functional-test-bundle/src/Test/ValidationErrorsConstraint.php:33
/vendor/liip/functional-test-bundle/src/Utils/HttpAssertions.php:101
/vendor/liip/functional-test-bundle/src/Test/WebTestCase.php:409
/tests/AppBundle/Controller/Api/PatientControllerTest.php:19

ERRORS!
Tests: 1, Assertions: 5, Errors: 1.

Remaining direct deprecation notices (3)

Remaining indirect deprecation notices (3)
```

--- a/composer.json
+++ b/composer.json
@@ -17,13 +17,13 @@
     "require": {
         "php": "^7.1",
         "doctrine/annotations": "^1.3",
+        "phpunit/phpunit": "^7.5.0 || ^8.0",
         "symfony/browser-kit": "^3.4 || ^4.1",
         "symfony/framework-bundle": "^3.4 || ^4.1"
     },
     "require-dev": {
         "doctrine/doctrine-bundle": "^1.8",
         "doctrine/orm": "^2.6",
-        "phpunit/phpunit": "^7.5.0 || ^8.0",
         "symfony/monolog-bundle": "^3.2",
         "symfony/phpunit-bridge": "^3.4 || ^4.1",
         "symfony/symfony": "^3.4 || ^4.1",

--- a/composer.json
+++ b/composer.json
@@ -23,8 +23,8 @@
     "require-dev": {
         "doctrine/doctrine-bundle": "^1.8",
         "doctrine/orm": "^2.6",
-        "phpunit/phpunit": "^6.1 || ^7.0",
-        "symfony/monolog-bundle": "^3.1",
+        "phpunit/phpunit": "^7.5.0 || ^8.0",
+        "symfony/monolog-bundle": "^3.2",
         "symfony/phpunit-bridge": "^3.4 || ^4.1",
         "symfony/symfony": "^3.4 || ^4.1",
         "twig/twig": "^2.0"

--- a/doc/basic.md
+++ b/doc/basic.md
@@ -103,7 +103,7 @@ Get the content of an URL:
 $content = $this->fetchContent('/contact');
 
 // `filter()` can't be used since the output is HTML code, check the content directly
-$this->assertContains(
+$this->assertStringContainsString(
     '<h1>LiipFunctionalTestBundle</h1>',
     $content
 );

--- a/doc/examples.md
+++ b/doc/examples.md
@@ -77,7 +77,7 @@ class ExampleFunctionalTest extends WebTestCase
         $this->assertSame('Hello foo!', $content);
 
         // check if the logout button is shown
-        $this->assertContains('logout', $content);
+        $this->assertStringContainsString('logout', $content);
     }
 
     public function test404Page(): void
@@ -88,7 +88,7 @@ class ExampleFunctionalTest extends WebTestCase
     public function testLoginPage(): void
     {
         $content = $this->fetchContent('/', 'GET', false);
-        $this->assertContains('login', $content);
+        $this->assertStringContainsString('login', $content);
     }
 
     public function testValidationErrors(): void

--- a/src/Test/ValidationErrorsConstraint.php
+++ b/src/Test/ValidationErrorsConstraint.php
@@ -30,7 +30,6 @@ class ValidationErrorsConstraint extends Constraint
     {
         $this->expect = $expect;
         sort($this->expect);
-        parent::__construct();
     }
 
     /**

--- a/tests/Command/CommandConfigTest.php
+++ b/tests/Command/CommandConfigTest.php
@@ -42,10 +42,10 @@ class CommandConfigTest extends WebTestCase
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
 
         // Test values from configuration
-        $this->assertContains('Environment: test', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Environment: test', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
 
-        $this->assertInternalType('boolean', $this->getDecorated());
+        $this->assertIsBool($this->getDecorated());
         $this->assertFalse($this->getDecorated());
     }
 }

--- a/tests/Command/CommandTest.php
+++ b/tests/Command/CommandTest.php
@@ -31,11 +31,11 @@ class CommandTest extends WebTestCase
         $this->commandTester = $this->runCommand('liipfunctionaltestbundle:test');
 
         // Test default values
-        $this->assertContains('Environment: test', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Environment: test', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
         $this->assertFalse($this->commandTester->getInput()->isInteractive());
 
-        $this->assertInternalType('boolean', $this->getDecorated());
+        $this->assertIsBool($this->getDecorated());
         $this->assertTrue($this->getDecorated());
 
         // Run command and reuse kernel
@@ -44,8 +44,8 @@ class CommandTest extends WebTestCase
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
         $this->assertEquals(0, $this->commandTester->getStatusCode());
 
-        $this->assertContains('Environment: test', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Environment: test', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
     }
 
     public function testRunCommandWithInputs(): void
@@ -68,10 +68,10 @@ class CommandTest extends WebTestCase
         $this->assertEquals(0, $this->commandTester->getStatusCode());
 
         // Test default values
-        $this->assertContains('Environment: test', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Environment: test', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
 
-        $this->assertInternalType('boolean', $this->getDecorated());
+        $this->assertIsBool($this->getDecorated());
         $this->assertTrue($this->getDecorated());
 
         // Run command and not reuse kernel
@@ -80,8 +80,8 @@ class CommandTest extends WebTestCase
 
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
 
-        $this->assertContains('Environment: prod', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Environment: prod', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
     }
 
     public function testRunCommandWithoutDecoration(): void
@@ -94,9 +94,9 @@ class CommandTest extends WebTestCase
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
         $this->assertEquals(0, $this->commandTester->getStatusCode());
 
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
 
-        $this->assertInternalType('boolean', $this->getDecorated());
+        $this->assertIsBool($this->getDecorated());
         $this->assertFalse($this->getDecorated());
     }
 
@@ -106,7 +106,7 @@ class CommandTest extends WebTestCase
         $this->assertSame(OutputInterface::VERBOSITY_QUIET, $this->getVerbosityLevel());
 
         $this->isDecorated(false);
-        $this->assertInternalType('boolean', $this->getDecorated());
+        $this->assertIsBool($this->getDecorated());
         $this->assertFalse($this->getDecorated());
 
         $this->commandTester = $this->runCommand('liipfunctionaltestbundle:test');
@@ -115,10 +115,10 @@ class CommandTest extends WebTestCase
         $this->assertEquals(0, $this->commandTester->getStatusCode());
 
         $this->assertEmpty($this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: DEBUG', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: DEBUG', $this->commandTester->getDisplay());
     }
 
     public function testRunCommandVerbosityImplicitlyNormal(): void
@@ -127,7 +127,7 @@ class CommandTest extends WebTestCase
         $this->assertSame(OutputInterface::VERBOSITY_NORMAL, $this->getVerbosityLevel());
 
         $this->isDecorated(false);
-        $this->assertInternalType('boolean', $this->getDecorated());
+        $this->assertIsBool($this->getDecorated());
         $this->assertFalse($this->getDecorated());
 
         $this->commandTester = $this->runCommand('liipfunctionaltestbundle:test');
@@ -135,10 +135,10 @@ class CommandTest extends WebTestCase
 
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
 
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: DEBUG', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: DEBUG', $this->commandTester->getDisplay());
     }
 
     public function testRunCommandVerbosityExplicitlyNormal(): void
@@ -152,10 +152,10 @@ class CommandTest extends WebTestCase
 
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
 
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: DEBUG', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: DEBUG', $this->commandTester->getDisplay());
     }
 
     public function testRunCommandVerbosityVerbose(): void
@@ -168,10 +168,10 @@ class CommandTest extends WebTestCase
 
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
 
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: DEBUG', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: DEBUG', $this->commandTester->getDisplay());
     }
 
     public function testRunCommandVerbosityVeryVerbose(): void
@@ -180,7 +180,7 @@ class CommandTest extends WebTestCase
         $this->assertSame(OutputInterface::VERBOSITY_VERY_VERBOSE, $this->getVerbosityLevel());
 
         $this->isDecorated(false);
-        $this->assertInternalType('boolean', $this->getDecorated());
+        $this->assertIsBool($this->getDecorated());
         $this->assertFalse($this->getDecorated());
 
         $this->commandTester = $this->runCommand('liipfunctionaltestbundle:test');
@@ -188,10 +188,10 @@ class CommandTest extends WebTestCase
 
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
 
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
-        $this->assertNotContains('Verbosity level: DEBUG', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringNotContainsString('Verbosity level: DEBUG', $this->commandTester->getDisplay());
     }
 
     public function testRunCommandVerbosityDebug(): void
@@ -200,7 +200,7 @@ class CommandTest extends WebTestCase
         $this->assertSame(OutputInterface::VERBOSITY_DEBUG, $this->getVerbosityLevel());
 
         $this->isDecorated(false);
-        $this->assertInternalType('boolean', $this->getDecorated());
+        $this->assertIsBool($this->getDecorated());
         $this->assertFalse($this->getDecorated());
 
         $this->commandTester = $this->runCommand('liipfunctionaltestbundle:test');
@@ -208,10 +208,10 @@ class CommandTest extends WebTestCase
 
         $this->assertInstanceOf(CommandTester::class, $this->commandTester);
 
-        $this->assertContains('Verbosity level: NORMAL', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
-        $this->assertContains('Verbosity level: DEBUG', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: NORMAL', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: VERY_VERBOSE', $this->commandTester->getDisplay());
+        $this->assertStringContainsString('Verbosity level: DEBUG', $this->commandTester->getDisplay());
     }
 
     public function testRunCommandStatusCode(): void
@@ -223,12 +223,11 @@ class CommandTest extends WebTestCase
         $this->assertEquals(10, $this->commandTester->getStatusCode());
     }
 
-    /**
-     * @expectedException \OutOfBoundsException
-     */
     public function testRunCommandVerbosityOutOfBound(): void
     {
         $this->setVerbosityLevel('foobar');
+
+        $this->expectException(\OutOfBoundsException::class);
 
         $this->runCommand('liipfunctionaltestbundle:test');
     }

--- a/tests/Command/ParatestCommandTest.php
+++ b/tests/Command/ParatestCommandTest.php
@@ -47,12 +47,12 @@ class ParatestCommandTest extends WebTestCase
             'options' => 'Tests/Test/WebTestCaseTest.php',
         ]);
 
-        $this->assertContains('Running phpunit in 3 processes with vendor/bin/phpunit', $content);
-        $this->assertContains('Initial schema created', $content);
-        $this->assertNotContains('Error : Install paratest first', $content);
-        $this->assertContains('Done...Running test.', $content);
+        $this->assertStringContainsString('Running phpunit in 3 processes with vendor/bin/phpunit', $content);
+        $this->assertStringContainsString('Initial schema created', $content);
+        $this->assertStringNotContainsString('Error : Install paratest first', $content);
+        $this->assertStringContainsString('Done...Running test.', $content);
 
-        $this->assertContains(
+        $this->assertStringContainsString(
             'OK (22 tests, 69 assertions)',
             $content
         );

--- a/tests/Test/WebTestCaseConfigTest.php
+++ b/tests/Test/WebTestCaseConfigTest.php
@@ -145,8 +145,6 @@ class WebTestCaseConfigTest extends WebTestCase
      *
      * There will be 2 queries, in the configuration the limit is 1,
      * an Exception will be thrown.
-     *
-     * @expectedException \Liip\FunctionalTestBundle\Exception\AllowedQueriesExceededException
      */
     public function testAllowedQueriesExceededException(): void
     {
@@ -160,6 +158,8 @@ class WebTestCaseConfigTest extends WebTestCase
         // One another query to load the second user.
         $path = '/user/2';
 
+        $this->expectException(\Liip\FunctionalTestBundle\Exception\AllowedQueriesExceededException::class);
+
         $this->client->request('GET', $path);
     }
 
@@ -170,8 +170,6 @@ class WebTestCaseConfigTest extends WebTestCase
      *
      * There will be 1 query, in the annotation the limit is 0,
      * an Exception will be thrown.
-     *
-     * @expectedException \Liip\FunctionalTestBundle\Exception\AllowedQueriesExceededException
      */
     public function testAnnotationAndException(): void
     {
@@ -179,6 +177,8 @@ class WebTestCaseConfigTest extends WebTestCase
 
         // One query to load the second user
         $path = '/user/1';
+
+        $this->expectException(\Liip\FunctionalTestBundle\Exception\AllowedQueriesExceededException::class);
 
         $this->client->request('GET', $path);
     }

--- a/tests/Test/WebTestCaseTest.php
+++ b/tests/Test/WebTestCaseTest.php
@@ -67,7 +67,7 @@ class WebTestCaseTest extends WebTestCase
             ]
         );
 
-        $this->assertInternalType('string', $path);
+        $this->assertIsString($path);
 
         $this->assertSame($path, '/user/1?get_parameter=abc');
     }
@@ -152,9 +152,9 @@ class WebTestCaseTest extends WebTestCase
         try {
             $this->assertStatusCode(-1, $this->client);
         } catch (AssertionFailedError $e) {
-            $this->assertContains('No route found for "GET /9999"', $e->getMessage());
-            $this->assertContains('Symfony\Component\HttpKernel\EventListener\RouterListener->onKernelRequest(', $e->getMessage());
-            $this->assertContains('Failed asserting that 404 matches expected -1.', $e->getMessage());
+            $this->assertStringContainsString('No route found for "GET /9999"', $e->getMessage());
+            $this->assertStringContainsString('Symfony\Component\HttpKernel\EventListener\RouterListener->onKernelRequest(', $e->getMessage());
+            $this->assertStringContainsString('Failed asserting that 404 matches expected -1.', $e->getMessage());
 
             return;
         }
@@ -211,9 +211,9 @@ class WebTestCaseTest extends WebTestCase
 
         $content = $this->fetchContent($path);
 
-        $this->assertInternalType('string', $content);
+        $this->assertIsString($content);
 
-        $this->assertContains(
+        $this->assertStringContainsString(
             '<h1>LiipFunctionalTestBundle</h1>',
             $content
         );
@@ -285,7 +285,7 @@ EOF;
 
         $this->assertStatusCode(200, $this->client);
 
-        $this->assertContains(
+        $this->assertStringContainsString(
             'Name submitted.',
             $crawler->filter('div.flash-notice')->text()
         );
@@ -318,7 +318,7 @@ EOF;
 
         $this->assertStatusCode(200, $this->client);
 
-        $this->assertContains(
+        $this->assertStringContainsString(
             'Name submitted.',
             $crawler->filter('div.flash-notice')->text()
         );
@@ -326,8 +326,6 @@ EOF;
 
     /**
      * @depends testForm
-     *
-     * @expectedException \PHPUnit\Framework\ExpectationFailedException
      */
     public function testFormWithException(): void
     {
@@ -341,6 +339,8 @@ EOF;
         $this->client->submit($form);
 
         $this->assertStatusCode(200, $this->client);
+
+        $this->expectException(\PHPUnit\Framework\ExpectationFailedException::class);
 
         $this->assertValidationErrors([''], $this->client->getContainer());
     }

